### PR TITLE
Added podAnnotations

### LINKED ISSUE #427: Custom annotation to pods
**Describe the solution you'd like**

I wish I could add annotations to the provider `daemoset` metadata.
The chart provides the possibility only to add labels and not annotations.

https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/charts/csi-secrets-store-provider-azure/templates/provider-azure-installer.yaml#L15-L19

**Anything else you would like to add:**

Please? 🙏🏻

### PR DESCRIPTION
**Pull request by pierluigilenoci**
**Reason for Change**:
I want add annotations to the provider daemoset metadata.
The chart provides the possibility only to add labels and not annotations.

**Requirements**

- [X] squashed commits
- [X] included documentation
- [X] added unit tests and e2e tests (if applicable).

**Issue Fixed**:
Fixed #427 

**Does this change contain code from or inspired by another project?**

- [ ] Yes
- [X] No

**Special Notes for Reviewers**:

Thank you @aramase for your work

### CODE REVIEW

File: manifest_staging/charts/csi-secrets-store-provider-azure/templates/provider-azure-installer-windows.yaml [hunk 1]
```diff
@@ -16,6 +16,9 @@ spec:
 {{ include "sscdpa.labels" . | indent 6 }}
 {{- if .Values.windows.podLabels }}
 {{- toYaml .Values.windows.podLabels | nindent 8 }}
+{{- end }}
+{{- if .Values.linux.podAnnotations }}
+{{- toYaml .Values.linux.podAnnotations | nindent 8 }}
```
Comment (aramase) [inline manifest_staging/charts/csi-secrets-store-provider-azure/templates/provider-azure-installer-windows.yaml:21]:
```
{{- if .Values.windows.podAnnotations }}
  annotations:
{{- toYaml .Values.windows.podAnnotations | nindent 8 }}
{{- end }}
```

File: manifest_staging/charts/csi-secrets-store-provider-azure/templates/provider-azure-installer.yaml [hunk 2]
```diff
@@ -16,6 +16,9 @@ spec:
 {{ include "sscdpa.labels" . | indent 6 }}
 {{- if .Values.linux.podLabels }}
 {{- toYaml .Values.linux.podLabels | nindent 8 }}
+{{- end }}
+{{- if .Values.linux.podAnnotations }}
```
Comment (aramase) [inline manifest_staging/charts/csi-secrets-store-provider-azure/templates/provider-azure-installer.yaml:20]:
```
{{- if .Values.linux.podAnnotations }}
  annotations:
{{- toYaml .Values.linux.podAnnotations | nindent 8 }}
{{- end }}
```

**Review by aramase — CHANGES_REQUESTED** [review]

File: manifest_staging/charts/csi-secrets-store-provider-azure/templates/provider-azure-installer-windows.yaml (hunk 1 above)
Comment (aramase) [reply manifest_staging/charts/csi-secrets-store-provider-azure/templates/provider-azure-installer-windows.yaml:22]: The values here should be `.Values.windows`

**Review by aramase — COMMENTED** [review]
The windows yaml checks need to be updated to check `.Values.windows.podAnnotations` instead of `.Values.linux.podAnnotations`

File: manifest_staging/charts/csi-secrets-store-provider-azure/templates/provider-azure-installer-windows.yaml (hunk 1 above)
Comment (pierluigilenoci) [reply manifest_staging/charts/csi-secrets-store-provider-azure/templates/provider-azure-installer-windows.yaml]: Fixed this as well. I'm sorry but I'm a little tired today.

**Review by aramase — APPROVED** [review]

**Comment by pierluigilenoci** [comment]
Thanks!

**Comment by aramase** [comment]
Thank you for the PR, @pierluigilenoci!

**Pull request merged**